mit_d3m/__init__.py

```diff
@@ -11,63 +11,98 @@
 import tarfile
 
 import boto3
+import botocore
+import botocore.config
+from funcy import memoize
 
 from mit_d3m.dataset import D3MDS
 from mit_d3m.loaders import get_loader
 from mit_d3m.metrics import METRICS_DICT
+from mit_d3m.utils import contains_files
+
+__all__ = (
+    'DATA_PATH',
+    'BUCKET',
+    'load_d3mds',
+    'load_dataset',
+)
+
 
 DATA_PATH = 'data'
 BUCKET = 'd3m-data-dai'
 
 
-def download_dataset(bucket, dataset, root_dir):
-    client = boto3.client('s3')
+@memoize
```

Comment:
i understand your point, though i think the memory footprint of the s3 client pales in comparison to the multi-GB files that are downloaded in some cases

I think the new dependency should be a secondary concern, funcy is in particular an extremely lightweight dependency (no additional downstream dependencies, extremely stable)

Reply:
Yes, you're right that the footprint is very low, and that the dependency is small as well.
It's just that, in general, I prefer to avoid adding dependencies unless they are really needed.
Let's keep it there!